src/logic/run-substream.ts

```diff
@@ -122,7 +123,60 @@ async function getConfigVars(config: IConfigComponent) {
   }
 }
 
-export async function runSubstream(
+export async function runSetupSubstream(
```

Comment:
Both the `runSetupSubstream` and `runSubstream` could be encapsulated into a `substreams` component which will receive the config or the logger components and use them.

Reply:
good idea! done!